tests/unit/test_environment.py

```diff
@@ -8,109 +7,106 @@
+    assert example_plain_env.latitude == latitude
+    assert example_plain_env.longitude == longitude
 
 
-def test_set_elevation(example_env):
-    """Tests if the elevation is set correctly in the environment object. This
-    one basically set the elevation using the set_elevation() method and then
-    checks if the elevation is the same as the one in the example environment
-    object.
+@pytest.mark.parametrize("elevation", [(-200), (0), (200)])
+def test_elevation_set_elevation_saves_elevation(elevation, example_plain_env):
+    """Tests elevation is set correctly in the environment obj.
 
     Parameters
     ----------
-    example_env : rocketpy.Environment
-        Example environment object to be tested.
+    example_plain_env : rocketpy.Environment
     """
-    example_env.set_elevation(elevation=200)
-    assert example_env.elevation == 200
+    example_plain_env.set_elevation(elevation=elevation)
+    assert example_plain_env.elevation == elevation
 
 
-def test_set_topographic_profile(example_env):
-    """Tests the topographic profile in the environment object.
+@pytest.mark.parametrize(
+    "latitude, longitude, theoretical_elevation", [(46.90479, 8.07575, 1565)]
+)
+def test_location_set_topographic_profile_computes_elevation(
+    latitude, longitude, theoretical_elevation, example_plain_env
+):
+    """Tests elevation computation given topographic profile in the environment obj.
 
     Parameters
     ----------
-    example_env : rocketpy.Environment
-        Example environment object to be tested.
+    example_plain_env : rocketpy.Environment
+    latitude: float
+        The latitude in decimal degrees.
+    longitude: float
+        The longitude in decimal degrees.
     """
-    example_env.set_location(46.90479, 8.07575)
-    example_env.set_topographic_profile(
+    example_plain_env.set_topographic_profile(
         type="NASADEM_HGT",
         file="data/sites/switzerland/NASADEM_NC_n46e008.nc",
         dictionary="netCDF4",
     )
-    assert (
-        example_env.get_elevation_from_topographic_profile(
-            example_env.latitude, example_env.longitude
-        )
-        == 1565
+    computed_elevation = example_plain_env.get_elevation_from_topographic_profile(
+        latitude, longitude
     )
+    assert computed_elevation == theoretical_elevation
 
 
-def test_export_environment(example_env_robust):
+def test_environment_export_environment_exports_environment_json(example_spaceport_env):
     """Tests the export_environment() method of the Environment class.
 
     Parameters
     ----------
-    example_env_robust : rocketpy.Environment
-        Example environment object to be tested.
+    example_spaceport_env : rocketpy.Environment
     """
-    assert example_env_robust.export_environment(filename="environment") == None
+    assert example_spaceport_env.export_environment(filename="environment") == None
+    assert os.path.isfile("environment.json")
```

Comment:
Asserting the file exists is already something good!
What do you think about opening the file and checking if the results are still like the expected values? For example, checking if all the keys are available. I worried about generating an empty file.

Reply:
Nice suggestion. BTW this made me find a bug in the environment code.

At some point in Environment code (line 1321), `self.pressure` is set to be equal to `self.pressure_ISA` (which is, indirectly a `Function()`) or to `Function()` directly. This led `export_environment()` to set `atmospheric_model_pressure_profile` as the memory address of the Function() obj, not the expected `pressure_function()` output.

I've fixed it both tests and lib by changing it to 

```
(...)
 ma.getdata(
                self.pressure.get_source()(self.height)
            ).tolist(),
```

A similar issue was happening with:
```
    "atmospheric_model_wind_velocity_x_profile": "<function Function.set_source.<locals>.source_function at 0x112804720>",
    "atmospheric_model_wind_velocity_y_profile": "<function Function.set_source.<locals>.source_function at 0x112804040>"
```

You can check it in detail in the commit BUG: Fixes environment_export

Reply:
Thank you for your solution.
I think this method is not so used, therefore the bug was never captured before.
Good to see it's better now.